tests/test_integration.py

```diff
@@ -378,3 +378,362 @@ def test(self, soco):
+                pytest.fail(msg)
+        for track in que:
+            soco.add_item_to_sonos_playlist(track, test_pl)
+        return test_pl
+
+    def test_reverse_track_order(self, soco):
+        """ Test reversing the tracks in the Sonos playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        tracks = range(num_tracks)
+        new_pos = list(reversed(tracks))
+
+        args = {'object_id': test_pl.item_id,
+                'tracks': tracks,
+                'new_pos': new_pos}
+        change, count, update_id = soco.reorder_sonos_playlist(**args)
+        assert change == 0
+        assert count == num_tracks
+        assert update_id != 0
+        spl = soco.browse(ml_item=test_pl)
+        for s_item, q_item in zip(spl, reversed(soco.get_queue())):
+            assert s_item.resources[0].uri != q_item.resources[0].uri
+
+    def test_swap_first_two_items(self, soco):
+        """ Test a use case in doc string. Swapping the positions of the first
+            two tracks in the Sonos playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        tracks = [0, ]
+        new_pos = [1, ]
+
+        args = {'object_id': test_pl.item_id,
+                'tracks': tracks,
+                'new_pos': new_pos}
+        change, count, update_id = soco.reorder_sonos_playlist(**args)
+        assert change == 0
+        assert count == num_tracks
+        assert update_id != 0
+        spl = soco.browse(ml_item=test_pl)
+        que = soco.get_queue()
+        assert spl[0].resources[0].uri == que[1].resources[0].uri
+        assert spl[1].resources[0].uri == que[0].resources[0].uri
+        for s_item, q_item in zip(spl[2:], que[2:]):
+            assert s_item.resources[0].uri == q_item.resources[0].uri
+
+    def test_remove_first_track(self, soco):
+        """ Test removing first track from Sonos Playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        tracks = [0, ]
+        new_pos = [None, ]
+
+        args = {'object_id': test_pl.item_id,
+                'tracks': tracks,
+                'new_pos': new_pos}
+        change, count, update_id = soco.reorder_sonos_playlist(**args)
+        assert change == -1
+        assert count == num_tracks - 1
+        assert update_id != 0
+        spl = soco.browse(ml_item=test_pl)
+        que = soco.get_queue()[1:]
+        for s_item, q_item in zip(spl, que):
+            assert s_item.resources[0].uri == q_item.resources[0].uri
+
+    def test_remove_first_track_full(self, soco):
+        """ Test removing first track from Sonos Playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        tracks = [0] + range(num_tracks - 1)    # [0, 0, 1, ..., n-1]
```

Comment:
Does this fail if you go all the way up to num_tracks? Just curious as this might shed some light on how the Sonos devices executes this.